src/replica/replica.cpp

```diff
@@ -503,44 +503,10 @@ std::string replica::query_manual_compact_state() const
     return _app->query_compact_state();
 }
 
-const char *manual_compaction_status_to_string(manual_compaction_status status)
+manual_compaction_status::type replica::get_manual_compact_status() const
 {
-    switch (status) {
-    case kIdle:
-        return "idle";
-    case kQueuing:
-        return "queuing";
-    case kRunning:
-        return "running";
-    case kFinished:
-        return "finished";
-    default:
-        dassert(false, "invalid status({})", status);
-        __builtin_unreachable();
-    }
-}
-
-manual_compaction_status replica::get_manual_compact_status() const
-{
-    std::string compact_state = query_manual_compact_state();
-    // query_manual_compact_state will return a message like:
-    // Case1. last finish at [-]
-    // - partition is not manual compaction
-    // Case2. last finish at [timestamp], last used {time_used} ms
-    // - partition manual compaction finished
-    // Case3. last finish at [-], recent enqueue at [timestamp]
-    // - partition is in manual compaction queue
-    // Case4. last finish at [-], recent enqueue at [timestamp], recent start at [timestamp]
-    // - partition is running manual compaction
-    if (compact_state.find("recent start at") != std::string::npos) {
-        return kRunning;
-    } else if (compact_state.find("recent enqueue at") != std::string::npos) {
-        return kQueuing;
-    } else if (compact_state.find("last used") != std::string::npos) {
-        return kFinished;
-    } else {
-        return kIdle;
-    }
+    dassert_replica(_app != nullptr, "");
```

Comment:
dcheck_eq(_app, nullptr)

Reply:
`dcheck_eq` won't print `replica_name()`, I suppose using `dcheck_eq_replica` is better. 
If you agree with me, I will raise another pull request to update other three problems including it.

Reply:
OK.